src/main/scala/org/hammerlab/magic/rdd/LazyZippedWithIndexRDD.scala

```diff
@@ -0,0 +1,78 @@
+package org.hammerlab.magic.rdd
+
+import scala.reflect.ClassTag
+
+import org.apache.spark.{Partition, TaskContext}
+import org.apache.spark.rdd.RDD
+
+/** Copy of [[ZippedWithIndexRDDPartition]] that cannot be accessed outside spark package */
+class LazyZippedWithIndexRDDPartition(val prev: Partition, val startIndex: Long)
+  extends Partition with Serializable {
+
+  override val index: Int = prev.index
+}
+
+/**
+ * Slight modification of [[ZippedWithIndexRDD]] from Spark that does not run job when created,
+ * it only triggers job on action or computing partitions.
+ */
+class LazyZippedWithIndexRDD[T: ClassTag](private var rdd: RDD[T]) extends RDD[(T, Long)](rdd) {
+  private var startIndices: Array[Long] = null
+
+  val iteratorSize: Iterator[_] => Long = iter => {
```

Comment:
ah, I changed this in https://github.com/sadikovi/magic-rdds/pull/1 for #8 as well… any reason not to use `iter.size`?